test/spell/RelinkV2Test.sol

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {RelinkV2Eth, VaultLike, AxelarAdapterLike} from "./RelinkV2Eth.sol";
+
+import {D18} from "../../src/misc/types/D18.sol";
+import {IAuth} from "../../src/misc/interfaces/IAuth.sol";
+import {IERC20} from "../../src/misc/interfaces/IERC20.sol";
+import {CastLib} from "../../src/misc/libraries/CastLib.sol";
+
+import {PoolId} from "../../src/common/types/PoolId.sol";
+import {AssetId} from "../../src/common/types/AssetId.sol";
+import {ShareClassId} from "../../src/common/types/ShareClassId.sol";
+import {ISpokeGatewayHandler} from "../../src/common/interfaces/IGatewayHandlers.sol";
+
+import {IShareToken} from "../../src/spoke/interfaces/IShareToken.sol";
+
+import {IBaseVault} from "../../src/vaults/interfaces/IBaseVault.sol";
+
+import {
+    UpdateRestrictionType,
+    UpdateRestrictionMessageLib
+} from "../../src/hooks/libraries/UpdateRestrictionMessageLib.sol";
+
+import "forge-std/Test.sol";
+
+contract RelinkV2TestBase is Test {
+    uint256 public constant REQUEST_ID = 0;
+
+    RelinkV2Eth public spell;
+
+    function setUp() public {
+        vm.createSelectFork("https://ethereum-rpc.publicnode.com");
+
+        spell = new RelinkV2Eth();
+    }
+
+    function castSpell() internal {
+        // NOTE: In production, this would be done through Guardian.scheduleRely + ROOT.executeScheduledRely
+        // For testing, we simulate this by directly setting the spell as ward on ROOT
+        // wards mapping is at slot 0 for Auth contracts
+        bytes32 wardSlot = keccak256(abi.encode(address(spell), uint256(0)));
+        vm.store(address(spell.V2_ROOT()), wardSlot, bytes32(uint256(1)));
+
+        spell.cast();
+    }
+}
+
+contract RelinkV2TestIntegrity is RelinkV2TestBase {
+    uint256 preJaaaTotalSupply;
+
+    // Main integration test
+    function testSpellIntegration() public {
+        preJaaaTotalSupply = spell.JAAA_SHARE_TOKEN().totalSupply();
+
+        castSpell();
+
+        _checkLinkedVaults();
+        _checkShareBalances();
+        _checkInvestmentState();
+        _checkMembershipState();
+        _checkAxelarTransactionCannotBeReexecuted();
+        _checkSpellCannotBeCastedSecondTime();
+        _checkSpellAccessIsCleared();
+    }
+
+    function _checkLinkedVaults() internal view {
+        assertEq(
+            spell.JTRSY_SHARE_TOKEN().vault(spell.USDC_TOKEN()),
+            spell.JTRSY_VAULT_ADDRESS(),
+            "JTRSY_SHARE_TOKEN.vault mismatch"
+        );
+        assertEq(
+            spell.JAAA_SHARE_TOKEN().vault(spell.USDC_TOKEN()),
+            spell.JAAA_VAULT_ADDRESS(),
+            "JAAA_SHARE_TOKEN.vault mismatch"
+        );
+    }
```

Comment:
NIT

We could add a pre-spell test against v3 vaults, but seems to be overkill.